tests/integ/datasource/test_oracledb.py

```diff
@@ -0,0 +1,169 @@
+
+try:
+    import pandas  # noqa: F401
+
+    is_pandas_available = True
+except ImportError:
+    is_pandas_available = False
+
+
+pytestmark = [
+    pytest.mark.skipif(
+        "config.getoption('local_testing_mode', default=False)",
+        reason="feature not available in local testing",
+    ),
+    pytest.mark.skipif(
+        not is_pandas_available,
+        reason="pandas is not available",
+    ),
+    pytest.mark.skipif(
+        RUNNING_ON_JENKINS,
+        reason="SNOW-2089683: oracledb real connection test failed on jenkins",
+    ),
+]
+
+
+ORACLEDB_TABLE_NAME = "ALL_TYPE_TABLE"
+ORACLEDB_TABLE_NAME_SMALL = "ALL_TYPE_TABLE_SMALL"
+ORACLEDB_TEST_EXTERNAL_ACCESS_INTEGRATION = "snowpark_dbapi_oracledb_test_integration"
+
+
+def test_dbapi_oracledb(session):
+    df = session.read.dbapi(
+        create_connection_oracledb,
+        table=ORACLEDB_TABLE_NAME,
+        max_workers=4,
+        query_timeout=5,
+    )
+    Utils.check_answer(df, oracledb_real_data)
+
+
+@pytest.mark.parametrize(
+    "create_connection, table_name, expected_result",
+    [
+        (
+            create_connection_oracledb,
+            ORACLEDB_TABLE_NAME,
+            oracledb_real_data,
+        ),
+        (
+            create_connection_oracledb,
+            ORACLEDB_TABLE_NAME_SMALL,
+            oracledb_real_data_small,
+        ),
+    ],
+)
+@pytest.mark.parametrize("fetch_size", [1, 3])
+def test_dbapi_batch_fetch(
+    session, create_connection, table_name, expected_result, fetch_size, caplog
+):
+    with caplog.at_level(logging.DEBUG):
+        df = session.read.dbapi(
+            create_connection, table=table_name, max_workers=4, fetch_size=fetch_size
+        )
+        # we only expect math.ceil(len(expected_result) / fetch_size) parquet files to be generated
+        # for example, 5 rows, fetch size 2, we expect 3 parquet files
+        assert caplog.text.count("Retrieved file from parquet queue") == math.ceil(
+            len(expected_result) / fetch_size
+        )
+        assert df.order_by("ID").collect() == expected_result
+
+
+def test_type_conversion():
+    invalid_type = OracleDBType("ID", "UNKNOWN", None, None, False)
+    with pytest.raises(NotImplementedError, match="oracledb type not supported"):
+        OracledbDriver(create_connection_oracledb, DBMS_TYPE.ORACLE_DB).to_snow_type(
+            [invalid_type]
+        )
+
+
+def test_oracledb_driver_coverage(caplog):
+    oracledb_driver = OracledbDriver(create_connection_oracledb, DBMS_TYPE.ORACLE_DB)
+    conn = oracledb_driver.prepare_connection(oracledb_driver.create_connection(), 0)
+    assert conn.outputtypehandler == output_type_handler
+
+    oracledb_driver.to_snow_type(
+        [OracleDBType("NUMBER_COL", oracledb.DB_TYPE_NUMBER, 40, 2, True)]
+    )
+    assert "Snowpark does not support column" in caplog.text
+
+
+@pytest.mark.skipif(
+    RUNNING_ON_JENKINS, reason="Cannot connect to oracledb from jenkins"
+)
```

Comment:
we have already skipped RUNNING_ON_JENKINS in the beginning. Can this be removed?

Reply:
removed